### ISSUE TITLE
0.64.0 layouts/partials/post/excerpt.html displays images twice and in some cases too big

### ISSUE THREAD

**Issue by chrisrummel** [razonyang/hugo-theme-bootstrap#346]
**Describe the bug**
The list view shows strange behaviour with the new layouts/partials/post/excerpt.html
Some preview images have the correct size, and while usually there should be only the posts text, the image is displayed a second time. With other posts the preview image is not small but huge, plus it gets displayed a second time.

**Screenshots**
(author removed screenshots)

**expecteds behaviour**
(author removed screenshots)

**Hugo version:**
```shell
hugo v0.92.0-B3549403+extended linux/amd64 BuildDate=2022-01-12T08:23:18Z VendorInfo=gohugoio
```

**Comment by razonyang** [comment]
@chrisrummel That is because `v0.64.0` changed the summaries' selection order, see also https://github.com/razonyang/hugo-theme-bootstrap/issues/341#issuecomment-1013617239 last point. But it isn't  backward compatibility.

The issue is about the summary format, plain text or HTML. We can do this via `summary` parameter or manual splitting(`<!--more-->`), but it may be dull if there are a lot of posts. So I prefer defining a new parameter(TBD, may be `post.plainifyExcerpt`) that allow users to pick up the format.

@sagarbehere @chrisrummel What do you think?

**Comment by chrisrummel** [comment]
@razonyang I see, the cause is a year old bug in hugo and there is nothing I could do against it besides copying excerpt.html from themes to local and modifying it to remove all HTML?

`post.plainifyExcerpt` sounds reasonable then, I guess.

**Comment by chrisrummel** [comment]
I tested
``
{{- else -}}
  {{- $readMore = false -}}
  {{- $maxLength := default 320 $.Site.Params.post.excerptMaxLength -}}
  {{- $content = .Content | plainify | truncate $maxLength $readMoreButton -}}
``
but that gives me unrendered htmlspecialchars.

**Comment by razonyang** [comment]
> I tested `{{- else -}}. {{- $readMore = false -}}. {{- $maxLength := default 320 $.Site.Params.post.excerptMaxLength -}}. {{- $content = .Content | plainify | truncate $maxLength $readMoreButton -}}` but that gives me unrendered htmlspecialchars.

Use `.Plain` instead of `.Content` may works.

**Comment by chrisrummel** [comment]
> Use `.Plain` instead of `.Content` may works.

No, sadly it doesn't work.

**Comment by sagarbehere** [comment]
@chrisrummel 

- Is [this](https://github.com/gohugoio/hugo/issues/8732) the issue you are getting when using `plainify`? If so, perhaps using `htmlUnescape` and `safeHTML` would help? 
- Another thing I can think of is just to use .Summary instead of .Content. That is how the HBS template originally did it. But this would mean that
    - You'll need to override the templates `layout/partials/post/excerpt.html` in your own site root dir
    - When [this Hugo issue](https://github.com/gohugoio/hugo/issues/7926) is fixed, this will stop working
 - Re: getting two images instead of one: Are you also using the `images` frontmatter variable? I was getting two images when I was using that variable AND the summary was not stripped of HTML

@razonyang I think `post.plainifyExcerpt` makes sense for those users who really do want to strip the HTML/Images/formatting from their summaries.

**Comment by razonyang** [comment]
@chrisrummel Could you please provide an example of content and expected screenshot for testing? Because you removed the screenshots, I can't test it by myself. You can send it to me by [e-mail](mailto:razonyang@gmail.com).
I also remember that there is a table(not plain text) inside a summary shown in the screenshot you posted, but I'm not sure if that is expected.

**Comment by chrisrummel** [comment]
@razonyang You can find a link to my site in the USERS.md and have a look at it.

**Comment by razonyang** [comment]
Closed by #351.

@sagarbehere FYI, the `post.plainifyExcerpt` was set as `true` by default, so that users won't be confused after upgrading the theme to v0.64.0 or above.

**Comment by chrisrummel** [comment]
@razonyang The ``htmlUnescape`` seems to have an unwanted sideeffect - the ``printf "<p>%s</p>"`` paragraph-tags show in the excerpt.

**Comment by chrisrummel** [comment]
> @chrisrummel It should be fixed by [f429db8](https://github.com/razonyang/hugo-theme-bootstrap/commit/f429db866d129398bffb69cd8a19cc0a47977ad8)

Indeed it is. Thanks a lot!
